test/Bart/Primitives/StringsTest.php

```diff
@@ -43,6 +43,48 @@ public function dataProviderTestSummarizeWithSuffix()
 		];
 	}
 
+    public function dataProviderTestStartsWith()
+    {
+        return [
```

Comment:
Try one where the length of substring is greater than full string when full string is not null. 
Also, one in which they are both null. 
Also try some invalid inputs like numbers.